### PR TITLE
RPC fixes

### PR DESCRIPTION
**Pull request by jagerman**
Various fixes to current dev RPC to get the testnet observer functioning again.

### CODE REVIEW

**Comment by jagerman** [comment]
There's probably more to do here, but I think this is good enough to merge into dev with anything else in follow-up PRs.

File: src/rpc/core_rpc_server.cpp [hunk 1]
```diff
@@ -1507,7 +1553,7 @@ namespace cryptonote::rpc {
     get_block.response["block_header"] = header;
     std::vector<std::string> tx_hashes;
     tx_hashes.reserve(blk.tx_hashes.size());
-    std::transform(blk.tx_hashes.begin(), blk.tx_hashes.end(), tx_hashes.begin(), [](const auto& x) { return tools::type_to_hex(x); });
+    std::transform(blk.tx_hashes.begin(), blk.tx_hashes.end(), std::back_inserter(tx_hashes), [](const auto& x) { return tools::type_to_hex(x); });
     get_block.response["tx_hashes"] = tx_hashes;
```
Comment (tewinget) [inline src/rpc/core_rpc_server.cpp:1557]: `std::move`?

**Review by darcys22 — APPROVED** [review]

File: src/rpc/core_rpc_server.cpp (hunk 1 above)
Comment (jagerman) [reply src/rpc/core_rpc_server.cpp]: Changed, though it probably doesn't do anything because of the vector<string> -> json conversion going on here.

Comment (tewinget) [reply src/rpc/core_rpc_server.cpp]: fair enough, but shouldn't hurt

**Comment by jagerman** [comment]
> There's probably more to do here, but I think this is good enough to merge into dev with anything else in follow-up PRs.

Famous last words.

So now, well, I think there's probably more to do here, but I think this is good enough to merge into dev with anything else in follow-up PRs.

**Pull request merged**